src/context.rs

```diff
@@ -63,28 +67,49 @@ impl Context {
         }
     }
 
+    /// Increases the reference count on self by one. This prevents self from being destroyed
+    /// until a matching call to drop() is made.
+    /// 
+    /// The number of references to a cairo_t can be get using Context::get_reference_count().
```

Comment:
It's quite complicated since the cairo API is quite old now. I think a big work on updating and finishing this crate is needed. For now, I just hide get_ptr functions.